src/bot/commands/admin-add-category.js

```diff
@@ -0,0 +1,47 @@
+import {query} from '../../services/db';
+import {createCommand, createParser} from 'chatter';
+import {parseMatches, prepareAddOutput, throwIfMatchErrors} from '../lib/matching';
+
+export default createCommand({
+  name: 'admin add category',
+  description: 'Add a new skills category.',
+  usage: '<category name>',
+}, createParser(({args}, {bot, token, getCommand}) => {
+  const search = args.join(' ');
+  if (!search) {
+    return false;
+  }
+  const output = [];
+  return query.categoryByName({token, search})
+    // parse matches
+    .then(results => parseMatches(results, search))
+    // return the matches and output
+    .then(results => prepareAddOutput(results, token))
+    .then(results => {
+
+      // if matches do not exist, add new category
+      const {matches} = results;
+      if (matches.length === 0) {
+        return query.categoryInsert({token, name: search})
+          .then(() => {
+            output.push(`_You have successfully added "${search}"._`);
+            return {output};
```

Comment:
I assumed that the next function in the promise chain, `throwIfMatchErrors` needed to know about matches so it could throw if there were errors. After looking at that function, it might be better titled `throwIfErrors`!